src/transaction/transaction_manager.cpp

```diff
@@ -55,9 +56,44 @@ timestamp_t TransactionManager::ReadOnlyCommitCriticalSection(TransactionContext
   return commit_time;
 }
 
+bool TransactionManager::CheckConstraints(TransactionContext *txn,
+                                          std::vector<TransactionConstraint *> *txn_installed_constraints) {
+  auto iter = constraints_.Begin();
+  while (iter != constraints_.End()) {
+    TransactionConstraint &constraint = *iter;
+    if (constraint.InstallingTransactionId() == txn->TxnId().load()) {
+      txn_installed_constraints->push_back(&constraint);
+      constraint.SetEnforcing();
```

Comment:
I think it's cleaner if you defer the setting rather than setting and then later undoing.